File: GameManager.cpp
#include "GameManager.hpp"
#include <iostream>

GameManager::GameManager() : m_players(0), m_currentPlayerID(-1), m_isRun(false)
{
	std::cout << "GameManager создан" << std::endl;
}

bool GameManager::GetIsRun()
{
	return m_isRun;
}
void GameManager::SetIsRun(bool isRun)
{
	m_isRun = isRun;
}
void GameManager::AddPlayer(Player& player)
{
	m_players.push_back(&player);
	std::cout << "Игрок \"" << player.GetName() << "\" добавлен" << std::endl;
}
void GameManager::StartGame()
{
	std::cout << "Игра начата" << std::endl;
}
void GameManager::StopGame()
{
	std::cout << "Игра приостановлена" << std::endl;
}
void GameManager::ResumeGame()
{
	std::cout << "Игра возобновлена" << std::endl;
}
void GameManager::EndGame()
{
	std::cout << "Игра окончена" << std::endl;
}

File: lab5/java/GameManager.java
import java.util.ArrayList;

public class GameManager
{

	private boolean isRun;
	private ArrayList<Player> players;
	private int currentPlayerID;


	public GameManager()
    {
        isRun = false;
        players = new ArrayList<Player>();
        currentPlayerID = -1;
        System.out.println("GameManager is created");
    }

    public boolean getIsRun()
    {
        return isRun;
    }
	public void AddPlayer(Player player)
    {
        players.add(player);
        System.out.println("Player is added");
    }
	public void StartGame()
    {
        System.out.println("Game is started");
        WindowManager.DrawField(players.get(0).getMyField());
        WindowManager.DrawField(players.get(0).getEnemyField());
    }
	public void StopGame()
    {
       
    }
	public void ResumeGame()
    {
        
    }
	public void EndGame()
    {
        System.out.println("Game is over");
    }   
}
